src/main/scala/scalafix/internal/sbt/SemanticRuleValidator.scala

```diff
@@ -10,13 +10,15 @@ class SemanticRuleValidator(ifNotFound: SemanticdbNotFound) {
   def findErrors(
       files: Seq[Path],
       dependencies: Seq[ModuleID],
+      scalacOpts: Seq[String],
       interface: ScalafixInterface
   ): Seq[String] = {
     if (files.isEmpty) Nil
     else {
       val errors = ListBuffer.empty[String]
       val hasSemanticdb =
-        dependencies.exists(_.name.startsWith("semanticdb-scalac"))
+        dependencies.exists(_.name.startsWith("semanticdb-scalac")) ||
+          scalacOpts.contains("-Xsemanticdb")
```

Comment:
this will bring a false positive on [early dotty & 3.0.0-M compilers](https://github.com/sbt/sbt/blob/91704c755c5f0c34d8cb9a9e3ef8b819c9ac344f/main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala#L46), but I guess it's fine